doc/user_manual/advanced_users_templates.tex

```diff
@@ -0,0 +1,167 @@
+\begin{lstlisting}[language=python]
+  def createWorkflow(self, model=None, variables=None, samples=None, case=None, **kwargs):
+\end{lstlisting}
+In order to correctly modify the Templated Workflow, the Template Class needs to know about what model is being sampled, the input variables to the model and how they're distributed, how many Monte Carlo samples to take, and the name of the case being run. It requires all of these to be provided by the Template Interface in order to write a new RAVEN input. In this case, the method arguments \texttt{model} and \texttt{variables} are dictionaries, while the \texttt{samples} are an integer and the \texttt{case} is a string.
+\\
+
+Note that \texttt{UQTemplate.createWorkflow} calls the Template Base Class's implementation first in order to preserve inheritance. Since the deepcopy happens in the base class, we don't perform it again in the custom Template Class.
+\\
+
+Throughout the remainder of the workflow creation, a series of XML manipulations are performed based on the inputs provided from the Template Interface. For example, the module to load for the Model is changed, the working directory is set, and the input and output variables are propagated throughout the input file. Note also that several input construction shortcut methods have been added for this particular template to simplify maintenance of the template.
+
+
+\subsubsection{Example Template Interface}
+The file \texttt{uq\_maker.py} contains the basic logic needed to read a user input file, load the Template Class, and generate new inputs. It follows the sequence of events outlined above, first instructing the Template Class to load the template, then reading in the user input, then instructing the Template Class to create the workflow, then to write the workflow.
+\\
+
+Beause the input needs for this Template are simple, we use Python's standard library \texttt{configparser} to read in the user input file, \texttt{uq\_template\_input.i}. This simple input structure uses sections (model, variables, and settings) with keyword and value pairs in each section. In order to change the RAVEN workflow created, the user only needs to make changes to the existing input file and run the interface, then run RAVEN on the new input.
+\\
+
+Note that we chose to provide the information to the Template Class mostly through dictionaries, where the essential pieces of information can be provided. In particular note that the variables provide only a mean and standard deviation; one reduction in flexibility is that we assume the variables are normally distributed, disallowing other distributions.
+\\
+
+The Template can be run with Python from the command line:
+\begin{lstlisting}[language=bash]
+ > python uq_maker.py
```

Comment:
It could be. It doesn't have to be, and as implemented, it's not that way. This particular test presumes a user that isn't going to write his own input file from scratch, but rather modify an existing input. The use of the "case" variable in the input allows the same input to be used over and over rather than writing many individual ones.

While this approach has tradeoffs with reading from an input file, this is all just for a test and demonstration case, so I don't feel there's a significant benefit one way or another.